src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
@davidsh, how important do you think these timeouts are?  I'm wondering if I should close this and we just won't expose them, or if I should spend more time working through this stuff.  Even sticking to the letter of the meaning that I believe WinHttp employs, Jeremy's right that it doesn't seem to provide much real value, e.g. if a server sent 1 byte every 30 seconds, satisfying the timeout but leading to ridiculously long transfers that would end up triggering overall timeouts.  I'm also wondering in what circumstances someone would really care about the distinction between SendTimeout, ReceiveHeadersTimeout, and ReceiveDataTimeout... i.e. if we do want to expose something, maybe it should just be SendReceiveTimeout (and on Windows that value would be passed to all of them)... it looks like WinHttpHandler already currently defaults to having them all be the same: https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs#L77-L79